### PR TITLE
EMCS-262: query parameters for lastUpdated/createdOn

### PR DESCRIPTION
**Pull request by Froom2**
Adds the query parameters to the get messages and get movements endpoints, to filter by date and time.
Inputting the incorrect format of date will return a BAD REQUEST response to the user.

### CODE REVIEW

**Comment by mpbostock** [comment]
> Adds the query parameters to the get messages and get movements endpoints, to filter by date and time. Inputting the incorrect format of date will currently result in an exception being thrown: an improvement ticket will be made to handle that scenario specifically.

Has this improvement ticket been raised?

File: test/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesControllerSpec.scala [hunk 1]
```diff
@@ -61,42 +62,111 @@ class GetMessagesControllerSpec extends PlaySpec
...
+      val result = createWithSuccessfulAuth.getMessagesForMovement(lrn, Some(timeNow))(createRequest())
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe Json.toJson(Seq(message))
+    }
+
+    "get all the new messages including messages with a createdOn time of NOW when there is a time query parameter provided" in {
+      val timeNow = dateTimeService.timestamp()
+
+      //TODO: do we need to do this truncated stuff everywhere? Or even at all. Just make sure that the format we have
```
Comment (mpbostock) [inline test/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesControllerSpec.scala:118]: Can this TODO now be removed?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala [hunk 2]
```diff
@@ -37,25 +38,36 @@ class GetMessagesController @Inject()(
...
         }
       }
     }
   }
 
-  private def getMessagesAsJson(lrn: String, ern: String): Future[Result] = {
-    movementService.getMovementByLRNAndERNIn(lrn, List(ern)).map {
-      case Some(mv) => Ok(Json.toJson(mv.messages))
+  private def getMessagesAsJson(lrn: String, ern: String, updatedSince: Option[String]): Future[Result] = {
+    val updatedSinceTime = updatedSince.map(Instant.parse(_))
```
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala:56]: What happens if the Instant isn't parsed properly, I can't see the error handling code, is this converted to an Internal Server Error or is the Exception shown to the user? If it fails to parse we should probably return a BadRequest with our standard Error Response?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/filters/MovementFilter.scala [hunk 3]
```diff
@@ -75,6 +84,7 @@ object MovementFilter {
             case "ern" => FilterErn(o._2)
             case "lrn" => FilterLrn(o._2)
             case "arc" => FilterArc(o._2)
+            case "updatedSince" => FilterUpdatedSince(o._2.map(Instant.parse(_)))
```
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/filters/MovementFilter.scala:87]: What happens if the Instant isn't parsed properly, I can't see the error handling code, is this converted to an Internal Server Error or is the Exception shown to the user? If it fails to parse we should probably return a BadRequest with our standard Error Response?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala (hunk 2 above)
Comment (Froom2) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala:56]: I'll do this instead yeah. Might as well do it now.

**Review by mpbostock — APPROVED** [review]

**Pull request merged**